recipe/build.sh

```diff
@@ -3,15 +3,43 @@
 mkdir build
 cd build
 
+# OS X
+# ----
+#
+# On MacOSX, the SDK is not distributed by conda
+#
+# Until version 10.13 (included), the OSX SDK headers were in
+# /usr/include/ and /usr/include/CXX_VERSION for the standard
+# library headers.
+#
+# From version 10.14, the OSX SDK is installed in
+# /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/
+#
+# Linux
+# -----
+#
+# The host compiler is distributed as a conda package.
+#
+
+EXTRA_CMAKE_ARGS=""
+if [[ `uname` == "Darwin" ]]; then
+  EXTRA_CMAKE_ARGS="${EXTRA_CMAKE_ARGS} -DCLING_INCLUDE_PATHS=/usr/include:/usr/include/c++/$(${CXX} -dumpversion):/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include:/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1"
```

Comment:
I think this will need to be changed slightly before this is merged but I'm still investigating.

Reply:
Thanks! I am currently travelling with a very slow connection, so I won't be able to test more thoroughly before a few hours.

Let me know what you find out.

Reply:
Definitely do not include absolute paths to the system here.

The SDK root should use `$CONDA_BUILD_SYSROOT`, which will be `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk` while 10.9 is being targeted. Including `/Library/Developer/...` should probably not be included, either. And also probably don't use /usr/include, which is likely to include things from the wrong version.

Reply:
@minrk These are paths used at runtime by cling.

Since there is no conda package for the OSX SDK, we must use the one at hand...

Reply:
Ah. hardcoding search paths at compile time that should be used at runtime is pretty much guaranteed to be buggy, since the right answer is not consistent across systems. As long as these are never used during the compilation of cling itself, you might be okay with a list in order of desirability:

```
# prefer target sdk if available
$CONDA_BUILD_SYSROOT/usr/include
# default sdk if deployment target not found, this will be present for any xcode install:
$(xcode-select -p)/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include
# system-wide default for old macOS:
/usr/include
```

Where does this get stored/used? Is there an environment variable you can set more reliably on activate?

Reply:
The `include/c++/v1` directories aren't needed and are what causes the ABI warning in https://github.com/conda-forge/cling-feedstock/pull/19#issuecomment-454320038. The warnings I was seeing are caused by cling not identifying the headers as being system headers. Launching the current anaconda.org build of cling like this on macOS 10.14 works without any warnings:

```bash
cling -isystem (xcode-select -p)/SDKs/MacOSX.sdk/usr/include
```

Reply:
[@minrk's suggestion](https://github.com/conda-forge/cling-feedstock/pull/19#discussion_r247874154) seems like the way to go. I'm asking the developers at the moment if there is an environment variable we can use (unlikely), else we could just patch the source.

Reply:
The place the include paths are used is [here](https://github.com/root-project/cling/blob/d713ae033e49fae9d34f88607a254033b8c4e8c0/lib/Interpreter/CIFactory.cpp#L649-L653).

We could patch the source at this same location to also populate runtime include directories set from an environment variable, but it does not look that robust either.

Reply:
> The `include/c++/v1 directories` aren't needed.

So we should remove them in that patch?

Reply:
> The `include/c++/v1` directories aren't needed.

Similarly (for OSX < 10.14), is the standard hearders directory `/usr/include/c++/$(${CXX} -dumpversion)` needed?

Reply:
We can use the `SDKROOT` environment variable to control where clang looks on macOS. We could also patch `clangdev` to respect `CONDA_BUILD_SYSROOT` by [duplicating this line](https://github.com/root-project/root/blob/master/interpreter/llvm/src/tools/clang/lib/Driver/ToolChains/Darwin.cpp#L1153).

Alternatively, `Toolchain/Darwin.cpp` could be patched to try `$(xcode-select -p)/SDKs/MacOSX.sdk/usr/include` to avoid the need for an environment variable.

> So we should remove them in that patch?

Yes

> Similarly (for OSX < 10.14), is the standard hearders directory /usr/include/c++/$(${CXX} -dumpversion) needed?

No they're distributed with conda and it already picks them up correctly.